core/domain/stats_jobs_one_off.py

```diff
@@ -246,65 +245,55 @@ def prepare_map(cls, item):
-                    'id': item.id,
-                    'created_on': python_utils.UNICODE(item.created_on),
-                    'session_id': item.session_id
-                })
-        elif isinstance(item, stats_models.SolutionHitEventLogEntryModel):
-            return (
-                item.exp_id,
-                {
-                    'event_type': feconf.EVENT_TYPE_SOLUTION_HIT,
-                    'version': item.exp_version,
-                    'state_name': item.state_name,
-                    'id': item.id,
-                    'created_on': python_utils.UNICODE(item.created_on),
-                    'session_id': item.session_id
-                })
-        elif isinstance(
-                item, stats_models.StartExplorationEventLogEntryModel):
-            return (
-                item.exploration_id,
-                {
-                    'event_type': feconf.EVENT_TYPE_START_EXPLORATION,
-                    'version': item.exploration_version,
-                    'state_name': item.state_name,
-                    'id': item.id,
-                    'created_on': python_utils.UNICODE(item.created_on),
-                    'session_id': item.session_id
-                })
-        elif isinstance(
-                item, stats_models.ExplorationActualStartEventLogEntryModel):
-            return (
-                item.exp_id,
-                {
-                    'event_type': feconf.EVENT_TYPE_ACTUAL_START_EXPLORATION,
-                    'version': item.exp_version,
-                    'state_name': item.state_name,
-                    'id': item.id,
-                    'created_on': python_utils.UNICODE(item.created_on),
-                    'session_id': item.session_id
-                })
-        elif isinstance(
-                item, stats_models.CompleteExplorationEventLogEntryModel):
-            return (
-                item.exploration_id,
-                {
-                    'event_type': feconf.EVENT_TYPE_COMPLETE_EXPLORATION,
-                    'version': item.exploration_version,
-                    'state_name': item.state_name,
-                    'id': item.id,
-                    'created_on': python_utils.UNICODE(item.created_on),
-                    'session_id': item.session_id
-                })
-        # pylint: enable=too-many-return-statements
+
+        if (
+                isinstance(
+                    item, (
+                        stats_models.CompleteExplorationEventLogEntryModel,
+                        stats_models.StartExplorationEventLogEntryModel,
+                        stats_models.StateHitEventLogEntryModel))):
+            event_dict = {
+                'version': item.exploration_version,
+                'state_name': item.state_name,
+                'id': item.id,
+                'created_on': python_utils.UNICODE(item.created_on),
+                'session_id': item.session_id
+            }
+
+            if isinstance(
```

Comment:
Figure out the event type first, then construct the event_dict in one go. Don't construct a dict and modify it.

Also, I actually would highly recommend making a dict mapping instance type to event type. The code would end up being simpler and more declarative.

Reply:
> I actually would highly recommend making a dict mapping instance type to event type

@seanlip I don't get this comment, could you please elaborate a little more?

Reply:
```
class_name_to_event_type = {
    'StateHitEventLogEntryModel': feconf.EVENT_TYPE_STATE_HIT,
    ...
}

```
then when you return the object, do:

```
event_dict = {
    'event_type': class_name_to_event_type[...],
    ...
}
```

Reply:
Done.

Reply:
Not done. The point of what I suggested above is not to have nested if/else. With the dictionary, you can have just a single `event_dict = ...` and you don't need to duplicate the code.

Reply:
I've tried to do that, but I didn't find a way to extract the info from `item` to use as a key, because if I use `type(item)` then it will return a lot of other info with the classname. Am I missing something here?

Reply:
`.__classname__` perhaps. 

Did you try searching? Getting the class name of an object is a standard operation.

Reply:
I tried searching but I think I again used the wrong keyword.

Reply:
I think I have given you enough information, and with the info you're giving me, I don't know enough about what you have done in order to advise you further.

Reply:
> I tried searching but I think I again used the wrong keyword.

No, You got me wrong, I found the solution to get the class name, I was just telling you the reason why I didn't find it the first time. Sorry for the misunderstanding.